inject/src/main/java/io/micronaut/inject/qualifiers/MatchArgumentQualifier.java

```diff
@@ -59,17 +59,43 @@ public static <T> MatchArgumentQualifier<T> ofArgument(Argument<?> argument) {
         );
     }
 
-    public static <T> MatchArgumentQualifier<T> ofSuperVariable(Class<T> beanType, Argument<?> argument) {
-        Argument<?> superArgument = Argument.ofTypeVariable(argument.getType(), null, argument.getAnnotationMetadata(), argument.getTypeParameters());
+    /**
+     * Finds matches of a type with a generic of a higher kind types (types that extend the type or are equal to it).
```

Comment:
"higher-kinded types" are something else, java doesn't have them. please rename this

Reply:
I think the proper name for this would be "covariant" matching (`ArrayList` matches `? extends List`), while the lowerType variant below would be "contravariant" matching (`List` matches `? super ArrayList`)

Reply:
Please check